enhancements/storage/manila-csi-driver-operator.md

```diff
@@ -0,0 +1,221 @@
+---
+title: manila-csi-driver-integration
+authors:
+  - "@mfedosin"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-11-22
+last-updated: 2019-11-22
+status: implementable
+---
+
+# Manila CSI driver integration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This document describes [Manila](https://docs.openstack.org/manila/latest/) [CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md) driver integration to enable ReadWriteMany([RWX](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes)) volume access on OpenShift 4 on OpenStack.
+
+## Motivation
+
+[Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes), when running on OpenStack, are backed by Cinder, which doesn’t allow RWX access. Users have applications that require this access mode and want to run OpenShift on their OpenStack platforms. If we add Manila CSI driver support, users will be able to run more applications on OpenShift 4 on OpenStack. Also rolling upgrades for StatefulSets aren’t supported without RWX.
+
+### Goals
+
+- Manila CSI driver image is available in the OCP Quay account.
+
+- Manila CSI driver operator manages the driver's lifecycle is created.
+
+- Manila CSI driver operator image is available in the OCP Quay account.
```

Comment:
Is there an easy way how to detect Manila in openshift-cluster-storage-operator (i.e. in go and with as little dependencies as possible)? Because that's the part that's going to decide if it should install the driver or not.

Reply:
Or maybe the openshift-cluster-storage-operator checks just for OpenStack, that's easy, and runs manilla operator in all clusters. The operator then does whatever it needs to detect if there is Manila installed and installs the CSI driver if so.